Library/ViewModels/ThanksViewModel.swift

```diff
@@ -199,6 +199,39 @@ public final class ThanksViewModel: ThanksViewModelType, ThanksViewModelInputs,
       reward: $0.reward,
       checkoutData: $0.checkoutData
     ) }
+
+    // FB CAPI
+    let facebookCAPIUserEmail = self.configureWithDataProperty.signal
+      .skipNil()
+      .map { $0.checkoutData?.facebookCAPIUserEmail }
+
+    let checkoutTotal = self.configureWithDataProperty.signal
+      .skipNil()
+      .map { ($0.checkoutData?.revenueInUsd) }
```

Comment:
So you can combine these into a tuple and have one signal instead of two, I would suggest a call like this:

```
    _ = Signal.combineLatest(project,
                         self.configureWithDataProperty.signal.skipNil())
      .observeValues { project, configData in
        var userEmail: String?
        var checkoutUSDAmount = ""
        
        if let checkoutDataValues = configData.checkoutData {
          userEmail = checkoutDataValues.facebookCAPIUserEmail
          checkoutUSDAmount = "\(checkoutDataValues.revenueInUsd)" + "\(checkoutDataValues.addOnsMinimumUsd)"
          checkoutUSDAmount += String(describing: checkoutDataValues.shippingAmountUsd)
          checkoutUSDAmount += String(describing: checkoutDataValues.bonusAmountInUsd)
        }

        guard featureFacebookConversionsAPIEnabled(), project.sendMetaCapiEvents,
          let externalId = AppTrackingTransparency.advertisingIdentifier() else { return }

        _ = AppEnvironment
          .current
          .apiService
          .triggerCapiEventInput(
            input: .init(
              projectId: "\(project.id)",
              eventName: FacebookCAPIEventName.BackingComplete.rawValue,
              externalId: externalId,
              userEmail: userEmail,
              appData: .init(extinfo: ["i2"]),
              customData: .init(currency: Currency.USD.rawValue, value: checkoutUSDAmount)
            )
          )
      }
```

Reply:
got it. thanks for the example!